File: app/src/main/java/com/imangazalievm/bubbble/di/ApplicationComponent.java
package com.imangazalievm.bubbble.di;

import com.imangazalievm.bubbble.di.modules.ApplicationModule;
import com.imangazalievm.bubbble.di.modules.DataModule;
import com.imangazalievm.bubbble.domain.repository.CommentsRepository;
import com.imangazalievm.bubbble.domain.repository.FollowersRepository;
import com.imangazalievm.bubbble.domain.repository.ImagesRepository;
import com.imangazalievm.bubbble.domain.repository.ShotsRepository;
import com.imangazalievm.bubbble.domain.repository.UsersRepository;

import javax.inject.Singleton;

import dagger.Component;
import ru.caprice.cafe.di.modules.OkHttpInterceptorsModule;

@Singleton
@Component(modules = {ApplicationModule.class, DataModule.class, OkHttpInterceptorsModule.class})
public interface ApplicationComponent {

    ShotsRepository provideBookRepository();

    CommentsRepository provideCommentsRepository();

    UsersRepository provideUsersRepository();

    FollowersRepository provideFollowersRepository();

    ImagesRepository provideImagesRepository();

}

File: app/src/main/java/com/imangazalievm/bubbble/di/ApplicationComponent.kt
package com.imangazalievm.bubbble.di

import com.imangazalievm.bubbble.data.shots.CommentsRepository
import com.imangazalievm.bubbble.data.shots.ImagesRepository
import com.imangazalievm.bubbble.data.shots.ShotsRepository
import com.imangazalievm.bubbble.data.users.FollowersRepository
import com.imangazalievm.bubbble.data.users.UsersRepository
import com.imangazalievm.bubbble.di.modules.OkHttpInterceptorsModule
import com.imangazalievm.bubbble.presentation.global.mvp.ErrorHandler
import com.imangazalievm.bubbble.presentation.global.permissions.PermissionsManager
import com.imangazalievm.bubbble.presentation.global.resourcesmanager.ResourcesManager
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        AppModule::class,
        DataModule::class,
        OkHttpInterceptorsModule::class
    ]
)
interface ApplicationComponent {

    val provideBookRepository: ShotsRepository
    val provideCommentsRepository: CommentsRepository
    val provideUsersRepository: UsersRepository
    val provideFollowersRepository: FollowersRepository
    val provideImagesRepository: ImagesRepository

    val errorHandler: ErrorHandler
    val resourcesManager: ResourcesManager
    val permissionsManager: PermissionsManager

}